### PR TITLE
Reduce drag when working locally in development

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

* Bump dependencies and fix `Dockerfile` regarding mimemagic
  * Without these changes, I could not bring up the `web` container
* Make DSA container create the UR-APO and add a role to the web container
  * This role grants the logged in user (via `ROLES` env var) permission to use any created APOs when registering items.

Both of these changes are included to make it easier to create fixture data.

## How was this change tested?

CI, locally

## Which documentation and/or configurations were updated?

docker-compose

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -313,6 +313,9 @@ GEM
     mime-types (3.3.1)
       mime-types-data (~> 3.2015)
     mime-types-data (3.2021.0225)
+    mimemagic (0.3.10)
```
Comment (jcoyne) [inline Gemfile.lock:316]: why did you add mimemagic as a dependency?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (jmartin-sul) [reply Gemfile.lock:316]: from glancing at `Gemfile`, my take is that this came along for the ride with the other updates?

how would you feel about approving/merging this to keep things moving, and ticketing mimemagic cleanup for later?  or, are you worried this'll cause trouble when deployed?  i have no guess on that last point either way.

**Review by jmartin-sul — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (mjgiarlo) [reply Gemfile.lock:316]: @jmartin-sul it shouldn't take too long or be too hairy to pluck this out and try again, so nbd

Comment (jmartin-sul) [reply Gemfile.lock:316]: ah ok, cool

Comment (mjgiarlo) [reply Gemfile.lock]: @jmartin-sul @jcoyne done.

**Review by jmartin-sul — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (jmartin-sul) [reply Gemfile.lock]: approved again 🙂

Comment (mjgiarlo) [reply Gemfile.lock]: @jcoyne another look?

File: Dockerfile [hunk 2]
```diff
@@ -7,6 +7,8 @@ RUN apk add --update --no-cache \
   libxslt-dev \
   # needed for mysql2 dependency
   mariadb-dev \
+  # needed because mimemagic-a-palooza
+  shared-mime-info \
```
Comment (jcoyne) [inline Dockerfile:11]: Is this still needed since mimemagic has been removed?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (mjgiarlo) [reply Dockerfile:11]: apparently not. removing. thanks!

**Comment by mjgiarlo** [comment]
@jcoyne have another look?

File: docker-compose.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ services:
       NODE_ENV: development
       RAILS_LOG_TO_STDOUT: 'true'
       REMOTE_USER: blalbrit@stanford.edu
-      ROLES: sdr:administrator-role
+      ROLES: sdr:administrator-role;sdr:service-manager
```
Comment (jcoyne) [inline docker-compose.yml:16]: I don't understand why this change is necessary. This role isn't one we use in https://github.com/sul-dlss/argo/blob/main/app/models/user.rb#L9

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (mjgiarlo) [reply docker-compose.yml:16]: @jcoyne they're used here: https://github.com/sul-dlss/argo/blob/main/app/forms/apo_form.rb#L7 Without adding this role, if you create an APO as admin, you can't even *see* it in the "register items" view.

**Review by jcoyne — APPROVED** [review]
